akka-testkit/src/main/scala/akka/testkit/ExplicitlyTriggeredScheduler.scala

```diff
@@ -0,0 +1,86 @@
+package akka.testkit
+
+import java.util.concurrent.ThreadFactory
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicLong
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+
+import com.typesafe.config.Config
+
+import akka.actor.{ ActorSystem, Cancellable, Scheduler }
+import akka.event.LoggingAdapter
+
+/**
+ * For testing: scheduler that does not look at the clock, but must be progressed manually by calling `timePasses`.
+ *
+ * This is not entirely realistic: jobs will be executed on the test thread instead of using the `ExecutionContext`, but does
+ * allow for faster and less timing-sensitive specs..
```

Comment:
Couldn't we still have them run on whatever execution context they want, but block `time Passes` on the futures completion?

Reply:
I'll give that a try!